### DISCUSSION TITLE
累計作業時間のリセット機能について

### DISCUSSION THREAD

**Discussion by kani3camp** [sorarideblog/youtube-study-space#139] [General]
累計作業時間をリセットすることで便利に使えることは理解しましたが、あとは、具体的にどのように実現するかです。
ランクまでリセットしたいというニーズは少ないと思うので、既存の累計作業時間とは別に、一時的な累計作業時間のデータをユーザごとに追加で持たせようと考えています。
ユーザーがリセットできるのは「一時的な」累計作業時間の方です。

**Comment by kani3camp** [comment]
他にも決めておかないといけないことがあるので、順次ここのコメントに追記していきます。
また、どなたでもお気軽に書き込みください。

**Reply by white113** [reply]
はじめまして。活用させていただいております。いつもありがとうございます。
個人的な話になってしまい申し訳ないのですが、いまランクによって席の色が変わる仕組みですけれども言い換えれば席の色が決まっているといえると思います。
実はお気に入りの色がありなんとなくお気に入りの色の席の方がモチベーションやテンションが上がる気がしてランクをリセットしたいなと思ってしまいました。細かいのでスルーしてくださって構わないのですが、ランクに紐付けされない色設定、もしくはリセット可能にしてもらえたらありがたいです。

**Reply by kani3camp** [reply]
返信が遅くなり申し訳ありません！
書き込みありがとうございます。
ランクに紐づけされない色設定、ですね。これまでそういう考え方はなかったのでとても参考になります。

ここからは独り言のようなことを書きますが、
「ランクに紐づけされない色設定」機能を作るとなると、既存のランクの色とは違う（例えば模様を変えるなど）ものにするのがいいのではと思いました。
ランク機能はランク機能でとても需要のある機能だと思います。その「ランクを見分ける方法」として現在は色分けがされているので、それと「ランクに紐づけされない色設定」が混じるとランクの色なのか自由に設定した色なのかわからなくなってしまうため、何かしら工夫が要りそうです。
そういえば、ランク機能のほうでもissue #160 で議論しているとおり、色を分ける以外の方法も検討中です。
ランクを見分ける方法として色以外を使うように仕様を変更すれば、「ランクに紐づけされない色設定」は問題なく実装できそうです。

また、累計作業時間をランクもろとも完全にリセットすることについては、できれば実装したくないなと個人的には思いつつも、選択肢として用意しておくのはユーザーからすれば便利だなあとも思います。

@white113 さんの意見も考慮しながら、まずは #160 の検討を済ませたいところです...

ところで、席の見え方を変えてテンションを上げるという点では、単に席の背景色を変えるのではなく、たとえばそのユーザーの好きな画像を背景にするのは面白いのではないかと考えてます。（背景画像の設定機能は有料会員もしくは高ランクの人限定にするイメージです）
